### DISCUSSION TITLE
Making sure ‘cbc_cmd’ result is deterministic

### DISCUSSION THREAD

**Discussion by s4ever** [coin-or/pulp#421] [Q&A] [answered]
Hi, 

I am trying to solve a fairly complex MIP modelled in PuLP using cbc_cmd. I am using default parameters. I find that at times the results are not deterministic. I did some search in internet and I know one parameter that affect will be timelimit. 

I have few questions here

1. What are the parameters I should set to make sure the solution is deterministic and repeatable? 
2. any other steps to be done for deterministic solution? Such as deleting the Lp file, or restarting the environment etc for each run?
3. Why timelimit parameter should affect determinism for a smaller problem? For example, say if my problem is simple enough to solve well within the timelimit, why could each run have different results?

Thank you in advance for all your help!

**Comment by pchtsp** [comment, marked as answer]
@tkralphs I think this is more of a CBC question. I'd gladly transfer the conversation but the `transfer discussion` dialog shows no repo to choose from.

@s4ever take a look at this repo: https://github.com/coin-or/Cbc. That's the CBC main repo. They will probably be able to help you better.

**Reply by s4ever** [reply]
Thanks @pchtsp ! I will post it there.

**Reply by s4ever** [reply]
Just in case anyone else stumble upon this question: A new discussion is created in CBC repo:
https://github.com/coin-or/Cbc/discussions/379
